Assets/Mirror/Runtime/NetworkWriter.cs

```diff
@@ -28,7 +62,9 @@ public class NetworkWriter
         public byte[] ToArray()
         {
             writer.Flush();
-            return ((MemoryStream)writer.BaseStream).ToArray();
+            byte[] data = ((MemoryStream)writer.BaseStream).ToArray();
+            Recycle(this);
```

Comment:
this seems very risky. we can't assume that people don't call .ToArray() twice, e.g. for Debug.Log etc.

Reply:
Current one is even more unsafe, and calling ToArray twice would never be a good idea

Reply:
why is the current one unsafe?

Reply:
Current reusings are made in a way in which we have no guarantess what would happen if somebody tries to reuse NetworkWriter when it's still used, this makes it perfectly safe.